core/arch/arm/kernel/thread_spmc.c

```diff
@@ -404,24 +404,26 @@ void spmc_handle_partition_info_get(struct thread_smc_args *args,
 	if (is_my_uuid(args->a1, args->a2, args->a3, args->a4)) {
 		spmc_fill_partition_entry(fpi, endpoint_id,
 					  CFG_TEE_CORE_NB_CORE);
+		rc = 1;
 	} else if (IS_ENABLED(CFG_SECURE_PARTITION)) {
 		uint32_t uuid_array[4] = { 0 };
 		TEE_UUID uuid = { };
 		TEE_Result res = TEE_SUCCESS;
+		size_t count = (rxtx->size / sizeof(*fpi));
 
 		uuid_array[0] = args->a1;
 		uuid_array[1] = args->a2;
 		uuid_array[2] = args->a3;
 		uuid_array[3] = args->a4;
 		tee_uuid_from_octets(&uuid, (uint8_t *)uuid_array);
 
-		res = sp_find_session_id(&uuid, &endpoint_id);
+		res = sp_partition_info_get(fpi, &uuid, &count);
+		rc = count;
```

Comment:
It would make more sense to assign this after `res` has been checked.

Reply:
Ack